### PR TITLE
RUN-3038 Load plugin modules

### PR DESCRIPTION
**Pull request by VanessaSergeon**
Given a plugin entry in the app.json, the runtime will need to determine the folder with preload plugins. 

[Wiki to be moved to Core repo](https://github.com/openfin/Internal-Wiki/wiki/RVM-to-Runtime-Plugin-Query)

[Tests](https://testing-dashboard.openfin.co/#/app/sessions/api/completed/59cd51e27cb79f732d10c3f3)  9/12 failing tests pass in isolation.

[Manual tests PR](https://github.com/openfin/test_apps/pull/8)

### CODE REVIEW

File: src/browser/api/window.js [hunk 1]
```diff
@@ -2165,4 +2165,8 @@ function getElectronBrowserWindow(identity, errDesc) {
     return browserWindow;
 }
 
+function getIdentifier(preload) {
+    return preload.url ? preload.url : `${preload.name}-${preload.version}`;
```
Comment (rdepena) [inline src/browser/api/window.js]: @VanessaSergeon why is getIdentifier also defined here ?

Comment (VanessaSergeon) [reply src/browser/api/window.js]: Oh thanks for catching that duplication. Will import from preload_scripts.ts instead.

**Comment by rdepena** [comment]
👍

File: src/browser/preload_scripts.ts [hunk 2]
```diff
@@ -136,37 +152,66 @@ function fetchToCache(identity: Identity, preloadScript: PreloadInstance): Promi
...
+            action: 'query-plugin',
+            name: preloadPlugin.name,
+            version: preloadPlugin.version,
+            optional: preloadPlugin.optional,
+            sourceUrl
+        };
+
+        rvmMessageBus.publish(msg, (resp) => {
+            if (resp.payload.hasOwnProperty('path') && resp.payload.action === 'query-plugin') {
+                const scriptPath = `${resp.payload.path}\\${resp.payload.target}`;
```
Comment (datamadic) [inline src/browser/preload_scripts.ts]: do we need to check that the `resp.payload.path` does not already end in a `\`? In general we should probably use the node `path` utility for stuff like this

**Pull request merged**